### PR TITLE
Fix: duplicate cached key issue

### PR DESCRIPTION
**Pull request by leoraba**
- Fix a duplicate submitter_specimen_id issue across analysis
- Enhancement: cleaner log messages
- Enhancement: retry after a timeout is reached

### CODE REVIEW

File: src/cache/index.ts [hunk 1]
```diff
@@ -78,10 +80,12 @@ function saveCacheAnalysis(analysisList: Array<Analysis>): Promise<void> {
               scorpioVersion: analysis.lineage_analysis?.scorpio_version || '',
             };
 
-            await saveHash(`sample:${analysis.samples.at(0)?.submitterSampleId}`, hsetData);
+            await saveHash(
+              `${analysis.studyId}:${analysis.samples.at(0)?.submitterSampleId}`,
```
Comment (joneubank) [inline src/cache/index.ts]: 👍

File: src/services/index.ts [hunk 2]
```diff
@@ -21,10 +21,10 @@ export const startUpdateAnalysisPipeline = function (): Promise<void> {
 export const handleData = new Writable({
   objectMode: true,
   write(source: TsvColumns, _encoding, callback) {
-    getCacheByKey(`sample:${source.specimen_collector_sample_ID}`)
+    getCacheByKey(`${source.study_id}:${source.specimen_collector_sample_ID}`)
```
Comment (joneubank) [inline src/services/index.ts]: should there be a utility method for building this key that we use both when calling save and get?

**Review by joneubank — APPROVED** [review]
Looks good. One comment that's just a suggestion.

File: src/services/index.ts (hunk 2 above)
Comment (leoraba) [reply src/services/index.ts]: followed your suggestion, I then added a method to take care for Hash key format in a single place. Thanks.

**Pull request merged**